java-frontend/src/main/java/org/sonar/java/regex/ast/RegexSource.java

```diff
@@ -31,7 +31,7 @@
  */
 public class RegexSource {
 
-  private final List<LiteralTree> stringLiterals;
+  final List<LiteralTree> stringLiterals;
```

Comment:
I'm not sure it's good to have its visibility increased. I would prefer to have a getter which would return an unmodifiable list, such as:
```java
List<LiteralTree> stringLiterals() {
  return Collections.unmodifiableList(stringLiterals);
}
```
It would prevent any accidental modification of the list.